javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue2361Test.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2015-2016 Federico Tomassetti
+ * Copyright (C) 2017-2019 The JavaParser Team.
+ *
+ * This file is part of JavaParser.
+ *
+ * JavaParser can be used either under the terms of
+ * a) the GNU Lesser General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ * b) the terms of the Apache License
+ *
+ * You should have received a copy of both licenses in LICENCE.LGPL and
+ * LICENCE.APACHE. Please refer to those files for details.
+ *
+ * JavaParser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ */
+
+package com.github.javaparser.symbolsolver;
+
+import com.github.javaparser.OpenIssueTest;
+import com.github.javaparser.ParserConfiguration;
+import com.github.javaparser.StaticJavaParser;
+import com.github.javaparser.ast.expr.ObjectCreationExpr;
+import com.github.javaparser.ast.visitor.VoidVisitorAdapter;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.CombinedTypeSolver;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.ReflectionTypeSolver;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Where a ternary operation has numerical values as the resulting branches, binary numeric promotion (bnp) is applied.
+ * <br>
+ * <br>
+ * 5.6.2 describes the outcome of bnp given the inputs, quoted below:
+ * <ul>
+ *     <li>If either operand is of type double, the other is converted to double.</li>
+ *     <li>Otherwise, if either operand is of type float, the other is converted to float.</li>
+ *     <li>Otherwise, if either operand is of type long, the other is converted to long.</li>
+ *     <li>Otherwise, both operands are converted to type int.</li>
+ * </ul>
+ *
+ * @see <a href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-5.html#jls-5.6.2">https://docs.oracle.com/javase/specs/jls/se8/html/jls-5.html#jls-5.6.2</a>
+ * @see <a href="https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.25">https://docs.oracle.com/javase/specs/jls/se8/html/jls-15.html#jls-15.25</a>
+ * @see <a href="https://github.com/javaparser/javaparser/issues/2361">https://github.com/javaparser/javaparser/issues/2361</a>
+ */
+public class Issue2361Test {
+
+
+    @OpenIssueTest(issueNumber = {2361, 2435})
```

Comment:
The issue numbers are really neat :-)

Reply:
It would be nice to be able to use them in some way.

Maybe a "fun project idea" to search for all these annotations and use the github api to find out if the issue is open/closed..? 

Or maybe an open issue test is now green, post to gitter or the pr/issue itself..?

Or maybe just add a javadoc comment with a url link to the github issue that can be copy/pasted..?


For now though, it can just be manually edited into the url or searched for via the web ui ;)

Reply:
It is a fun project idea, but the label "fun project idea" is for stuff that is a good idea, but is not going to be part of JP itself :-)